sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +202,111 @@ def _can_handle_query(cls, *query):
         boolean
             answer as to whether client can service the query
         """
-        chkattr = ['Time', 'Instrument', 'SatelliteNumber']
+        chkattr = ["Time", "Instrument", "SatelliteNumber"]
         chklist = [x.__class__.__name__ in chkattr for x in query]
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value.lower() in ('xrs', 'goes'):
+            if x.__class__.__name__ == "Instrument" and x.value.lower() in (
+                "xrs",
+                "goes",
+            ):
                 return all(chklist)
         return False
+
+
+class SUVIClient(GenericClient):
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Returns a URL to the SUVI data for the specified date.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+
+        level : str
+            The level of the data.
+
+        Note
+        ----
+        This is level 1b data is not implemented.
+        """
+        base_url = "https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/"
+
+        if "wavelength" not in kwargs.keys() or not kwargs["wavelength"]:
+            raise ValueError("Queries must specify a wavelength.")
+        else:
+            wavelength = kwargs["wavelength"]
+        wavelength = wavelength.to(u.Angstrom, equivalencies=u.spectral())
+
+        if "level" in kwargs:
+            level = kwargs["level"]
+        else:
+            level = "l2"
+
+        if level == "l2":
+            base_url += f'l2/data/suvi-{level}-ci{int(wavelength.to("Angstrom").value):03}/'
```

Comment:
If this data source has a predictable URL pattern could you not use the [Scraper](http://docs.sunpy.org/en/stable/api/sunpy.util.scraper.Scraper.html#sunpy.util.scraper.Scraper)?

Reply:
I wasn't sure. The filenames depend on when the image was taken and so cannot be predicted. I request all of the filenames and then compare to the time range.

Reply:
That's exactly what Scraper does iirc. You give it a filename pattern and it works out which file names exist within your time.

Reply:
@Cadair  I was having similar trouble with `Scraper` - for the SUVI files the filenames aren't predictable and have two dates within each filename e.g.`dr_suvi-l2-ci171_g16_s20190302T000000Z_e20190302T000400Z_v1-0-0.fits` - how can you go about this using Scraper?